src/py42/clients/cases.py

```diff
@@ -0,0 +1,185 @@
+
+    def get_all(
+        self,
+        name=None,
+        status=None,
+        min_create_time=None,
+        max_create_time=None,
+        min_update_time=None,
+        max_update_time=None,
+        subject=None,
+        assignee=None,
+        page_number=1,
+        page_size=100,
+        sort_direction=u"asc",
+        sort_key=u"number",
+        **kwargs
+    ):
+        """Gets all cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status. `CaseStatus.OPEN` or `CaseStatus.CLOSED` Defaults to None.
+            min_create_time (str or int or float or datetime, optional): Filter results by case creation time, start time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            max_create_time (str or int or float or datetime, optional): Filter results by case creation time, end time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            min_update_time (str or int or float or datetime, optional): Filter results by last updated time, start time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            max_update_time (str or int or float or datetime, optional): Filter results by last updated time, end time.
+                 str format %Y-%m-%d %H:%M:%S. Defaults to None.
+            subject (str, optional): Filter results based on User UID of a subject of a case. Defaults to None.
+            assignee (str, optional): Filter results based on User UID of an assignee of a case. Defaults to None.
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 100.
+            sort_direction (str, optional): The direction on which to sort the response,
+                based on the corresponding sort key. `asc` or `desc`. Defaults to `asc`.
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number".
+                Available options are `name`, `number`, `createdAt`, `updatedAt`, `status`, `assigneeUsername`, `subjectUsername`.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of cases.
+        """
+
+        created_at = CasesClient._make_range(min_create_time, max_create_time)
+        updated_at = CasesClient._make_range(min_update_time, max_update_time)
+
+        return self._cases_service.get_all(
+            name=name,
+            status=status,
+            created_at=created_at,
+            updated_at=updated_at,
+            subject=subject,
+            assignee=assignee,
+            page_number=page_number,
+            page_size=page_size,
+            sort_direction=sort_direction,
+            sort_key=sort_key,
+            **kwargs
```

Comment:
I don't think any of the values from `kwargs` make it to the request, from what I can.. Is this needed?

Reply:
thanks, missed adding it in `.services.get_page`. 
Added `kwargs` because API supports filter by other fields which we have not supported, like 'isAssigned', `lastModifiedBy`, etc.

Also do we need a `get_page` method for cases?

Reply:
Ah just noticing that we don't have one. Yeah, it'd be nice to have, as the generators don't work too well in certain applications. Do you want to add it now or just create a story for it and add it later?

Reply:
Added.